lib/storage/table.go

```diff
@@ -99,15 +102,17 @@ func mustOpenTable(path string, s *Storage) *table {
 	fs.MustRemoveTemporaryDirs(bigSnapshotsPath)
 
 	// Open partitions.
-	pts := mustOpenPartitions(smallPartitionsPath, bigPartitionsPath, s)
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
I don't really like an idea for cancelling with context. It adds a lot of changes. Maybe we should keep it with simple stop chan?

Reply:
Using a stop chan will require one of the following:
- to wrap channel into local channel to avoid double-close issue
- to pass stop chan from partition to table and only close the channel in parent partition

These options seems to be harder to maintain to me rather than using ctx directly. Wdyt?

Reply:
I would support @f41gh7 opinion regarding channels.

> Using a stop chan will require one of the following

What about extending the [following method](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/b08cbd0400edabbb53cc8cc3330105fcfe472f3e/lib/storage/partition.go#L1185) with extra param to accept channel for canceling? We could pass `table.stop` in it, and should do the trick.